### PR TITLE
LPS-159408: Generate permissions endpoints in Object Entry 

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Add `GET` and `PUT` permissions endpoints in _rest-openapi.yaml_
- Implement `getPermissionCheckerGroupId` and `getPermissionCheckerResourceName` methods.

**GET METHOD RESPONSE:**

```
{
  "actions": {
    "get": {
      "method": "GET",
      "href": "http://localhost:8080/o/c/testobjects/46735/permissions"
    },
    "replace": {
      "method": "PUT",
      "href": "http://localhost:8080/o/c/testobjects/46735/permissions"
    }
  },
  "facets": [],
  "items": [
    {
      "actionIds": [
        "DELETE",
        "PERMISSIONS",
        "UPDATE",
        "VIEW"
      ],
      "roleName": "Owner"
    },
    {
      "actionIds": [
        "DELETE",
        "PERMISSIONS"
      ],
      "roleName": "Power User"
    }
  ],
  "lastPage": 1,
  "page": 1,
  "pageSize": 2,
  "totalCount": 2
}
```

image

**How to Reproduce it:**
- Deploy `object-rest-api` and `object-rest-impl`

**JIRA TICKET**
https://issues.liferay.com/browse/LPS-159408

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]
Awesome

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 1]
```diff
@@ -401,6 +401,19 @@ public void update(
...
+	protected Long getPermissionCheckerGroupId(Object id) throws Exception {
+		return contextCompany.getCompanyId();
+	}
+
+	@Override
+	protected String getPermissionCheckerResourceName(Object id)
+		throws Exception {
+
+		return ObjectDefinition.class.getName() + "#" +
+			_objectDefinition.getObjectDefinitionId(); // ObjectDefinition#44002
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:414]: should be deleted the comment?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 2]
```diff
@@ -401,6 +401,19 @@ public void update(
 		super.update(objectEntries, parameters);
 	}
 
+	@Override
+	protected Long getPermissionCheckerGroupId(Object id) throws Exception {
+		return contextCompany.getCompanyId();
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:406]: Is the `companyId` what we have to return? I have seen in the PermissionUtil class that the id returned by this method is being used to get the group

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java (hunk 1 above)
Comment (sergiojimcos) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:414]: Ups, sorry I haven't see that :sweat_smile:

**Review by ccorreagg — CHANGES_REQUESTED** [review]
Please, review my comments

**Comment by ccorreagg** [comment]
As we talked, it is necessary to expose the endpoints also for scoped objects

**Comment by ccorreagg** [comment]
It would be great if we test that the endpoint is returning the id properly. For that, please, have a look at the tests in cluded in the 
 `StructuredContentResourceTest` class that overrrides a base clase generated by the REST Builder. There are some tests that are already testing some endpoints, and there are some methods that are overriden to indicate which fields must be validated, for example `getAdditionalAssertFieldNames`. Maybe it is necessary to include the new field in some methods to validate the change. Please, have a look at it

**Comment by sergiojimcos** [comment]
Closing this Pr because it is a draft investigation

**Pull request closed without merging**